arch/rv32i/src/lib.rs

```diff
@@ -388,17 +388,10 @@ pub extern "C" fn _start_trap() {
 #[cfg(all(target_arch = "riscv32", target_os = "none"))]
 #[link_section = ".init"]
 #[export_name = "abort"]
-pub extern "C" fn abort() {
-    unsafe {
-        llvm_asm! ("
-            // Simply go back to the start as if we had just booted.
-            j    _start
-        "
-        :
-        :
-        :
-        : "volatile");
-    }
+#[naked]
+pub unsafe extern "C" fn abort() {
```

Comment:
I think even better we can just remove this entire function. I'm not sure if it was needed before or just a bad copy-and-paste on my part.

Reply:
Right, it really doesn't seem to be referenced anywhere. Given it's side effects when called are pretty significant, it's probably best to remove if we don't use it. Done.